src/backends/cuda/cuda_device.cpp

```diff
@@ -1023,14 +1026,18 @@ static constexpr auto required_cuda_version_major = 11;
 static constexpr auto required_cuda_version_minor = 7;
 static constexpr auto required_cuda_version = required_cuda_version_major * 1000 + required_cuda_version_minor * 10;
 
-static void initialize() {
+[[nodiscard]]
+static std::pair<int, int> initialize() {
```

Comment:
This is incorrect. If we create multiple devices, all devices other than the first will receive incorrect driver version and device count.